### PR TITLE
BranchStatusColumn should not be exposed to users

### PR DESCRIPTION
**Pull request by stephenc**
@reviewbybees (found during some other stuff)

### CODE REVIEW

File: src/main/java/jenkins/branch/BranchStatusColumn.java [hunk 1]
```diff
@@ -104,4 +106,19 @@ public boolean shownByDefault() {
...
+     */
+    @Extension
+    public static class DescriptorVisibilityFilterImpl extends DescriptorVisibilityFilter {
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean filter(Object context, Descriptor descriptor) {
+            return !(descriptor instanceof DescriptorImpl) || context instanceof MultiBranchProjectViewHolder.ViewImpl;
```
Comment (rsandell) [inline src/main/java/jenkins/branch/BranchStatusColumn.java]: won't `context instanceof MultiBranchProjectViewHolder.ViewImpl` hide all descriptors if the view is of that type?

Comment (rsandell) [inline src/main/java/jenkins/branch/BranchStatusColumn.java]: Or maybe that is the intention?

Comment (stephenc) [inline src/main/java/jenkins/branch/BranchStatusColumn.java]: either it is not `BranchStatusColumn.DescriptorImpl`, in which case we take no action... or it is  `BranchStatusColumn.DescriptorImpl`, in which case we hide it from everywhere except the `MultiBranchProjectViewHolder.ViewImpl` (which is non-user configurable so we could hide it here also, except there may be other things that rewrite and apply the filters, so to be safe we allow it for that non-user configurable view)

**Comment by rsandell** [comment]
:bee:

**Comment by raul-arabaolaza** [comment]
AFAIK :bee:

File: src/main/java/jenkins/branch/BranchStatusColumn.java (hunk 1 above)
Comment (jglick) [inline src/main/java/jenkins/branch/BranchStatusColumn.java]: Might be clearer if written as

```java
return !(descriptor instanceof DescriptorImpl &&
         !(context instanceof MultiBranchProjectViewHolder.ViewImpl));
```

or even

```java
if (descriptor instanceof DescriptorImpl &&
    !(context instanceof MultiBranchProjectViewHolder.ViewImpl)) {
    return false;
} else {
    return true;
}
```

Of course clearer and more efficient still would have been if `@Extension` had never been invented and SezPoz had been used as intended (`-source 8` for brevity):

```java
@DescriptorVisibilityFilterRegistration(type=DescriptorImpl.class) public static final DescriptorVisibilityFilter HIDE_ME_ELSEWHERE =
    (context, descriptor) -> context instanceof MultiBranchProjectViewHolder.ViewImpl;
```

**Pull request merged**